### PR TITLE
mark NFW from OOP critical

### PR DESCRIPTION
**Pull request by heejaechang**
currently, whenever OOP throws an exception, we show infobar saying "restart VS". whenever that happens, we report NFW. now those NFW will be marked as critical and we will stop reporting NFW after that in VS.

basically making critical NFW behavior same as fatal watson in VS. and management will treat critical NFW same as fatal watson.

also, now we will show callstack like code fix exception when info bar is shown

image

### CODE REVIEW

**Comment by heejaechang** [comment]
@ivanbasov @jinujoseph can you take a look. as we discussed.

File: src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs [hunk 1]
```diff
@@ -31,19 +31,19 @@ internal static class WatsonReporter
         /// Report Non-Fatal Watson
         /// </summary>
         /// <param name="exception">Exception that triggered this non-fatal error</param>
-        public static void Report(Exception exception)
+        public static void Report(Exception exception, bool critical = false)
```
Comment (ivanbasov) [inline src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs]: >bool critical = false [](start = 55, length = 21)

maybe enum is better

File: src/Workspaces/Remote/ServiceHub/Telemetry/WatsonReporter.cs [hunk 2]
```diff
@@ -39,19 +39,19 @@ public static void SetTelemetrySession(TelemetrySession session)
         /// Report Non-Fatal Watson
         /// </summary>
         /// <param name="exception">Exception that triggered this non-fatal error</param>
-        public static void Report(Exception exception)
+        public static void Report(Exception exception, bool critical = false)
```
Comment (ivanbasov) [inline src/Workspaces/Remote/ServiceHub/Telemetry/WatsonReporter.cs]: >bool critical = fals [](start = 55, length = 20)

same here and below: enum maybe better

File: src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs (hunk 1 above)
Comment (heejaechang) [reply src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs]: @ivanbasov  do you want enum with every thing? or just Critical and None?

Comment (ivanbasov) [reply src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs]: I think that we may want the enum on our side to differ from the enum on the Telemetry side. So, we should define the enum for our purposes. Currently, it should contain Critical and None only.

Comment (heejaechang) [reply src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs]: alright. will do!

Comment (heejaechang) [reply src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs]: @ivanbasov  done!

**Pull request merged**